packages/rest/src/utils/agent.ts

```diff
@@ -1,79 +1,196 @@
-  : path.join(__dirname, '../../../../network/genesis/local-genesis.txn')
+type ModulesWithoutTenants = Omit<ReturnType<typeof getAgentModules>, 'tenants'>
+
+export type RestRootAgent = Agent<ModulesWithoutTenants>
+export type RestRootAgentWithTenants = Agent<ModulesWithoutTenants & { tenants: TenantsModule<ModulesWithoutTenants> }>
+export type RestTenantAgent = TenantAgent<ModulesWithoutTenants>
+export type RestAgent = RestRootAgent | RestTenantAgent | RestRootAgentWithTenants
+
+export const getAgentModules = (options: {
+  autoAcceptConnections: boolean
+  autoAcceptProofs: AutoAcceptProof
+  autoAcceptCredentials: AutoAcceptCredential
+  autoAcceptMediationRequests: boolean
+  indyLedgers?: [IndyVdrPoolConfig, ...IndyVdrPoolConfig[]]
+  extraAnonCredsRegistries?: AnonCredsRegistry[]
+  multiTenant: boolean
+}) => {
+  const legacyIndyCredentialFormatService = new LegacyIndyCredentialFormatService()
+  const legacyIndyProofFormatService = new LegacyIndyProofFormatService()
+
+  const baseModules = {
+    connections: new ConnectionsModule({
+      autoAcceptConnections: options.autoAcceptConnections,
+    }),
+    proofs: new ProofsModule({
+      autoAcceptProofs: options.autoAcceptProofs,
+      proofProtocols: [
+        new V1ProofProtocol({
+          indyProofFormat: legacyIndyProofFormatService,
+        }),
+        new V2ProofProtocol({
+          proofFormats: [legacyIndyProofFormatService, new AnonCredsProofFormatService()],
+        }),
+      ],
+    }),
+    credentials: new CredentialsModule({
+      autoAcceptCredentials: options.autoAcceptCredentials,
+      credentialProtocols: [
+        new V1CredentialProtocol({
+          indyCredentialFormat: legacyIndyCredentialFormatService,
+        }),
+        new V2CredentialProtocol({
+          credentialFormats: [legacyIndyCredentialFormatService, new AnonCredsCredentialFormatService()],
+        }),
+      ],
+    }),
+    anoncreds: new AnonCredsModule({
+      registries: [new IndyVdrAnonCredsRegistry(), ...(options.extraAnonCredsRegistries ?? [])],
+      anoncreds,
+    }),
+    askar: new AskarModule({
+      ariesAskar,
+      multiWalletDatabaseScheme: AskarMultiWalletDatabaseScheme.ProfilePerWallet,
+    }),
+    mediator: new MediatorModule({
+      autoAcceptMediationRequests: options.autoAcceptMediationRequests,
+    }),
+    dids: new DidsModule({
+      registrars: [new KeyDidRegistrar(), new JwkDidRegistrar(), new PeerDidRegistrar(), new IndyVdrIndyDidRegistrar()],
+      resolvers: [
+        new WebDidResolver(),
+        new KeyDidResolver(),
+        new JwkDidResolver(),
+        new PeerDidResolver(),
+        new IndyVdrIndyDidResolver(),
+        new IndyVdrSovDidResolver(),
+      ],
+    }),
+  } as const
+
+  const modules: typeof baseModules & { tenants?: TenantsModule<typeof baseModules>; indyVdr?: IndyVdrModule } =
+    baseModules
+
+  if (options.multiTenant) {
+    modules.tenants = new TenantsModule({
+      sessionLimit: Infinity,
```

Comment:
Thats very scalable! wonder what happens when you exceed f32::MAX sessions :').

Reply:
Yeah the session implementation is a bit broken in Credo, I want to address it for 0.6, as it counts very weird so it's not a good indication of how much resources you're using